lib/spack/spack/cmd/blame.py

```diff
@@ -112,6 +113,23 @@ def dump_json(rows, last_mod, total_lines, emails):
     sjson.dump(result, sys.stdout)
 
 
+def repo_prefix(path):
+    if path.startswith(spack.paths.prefix):
+        return spack.paths.prefix
+
+    descriptors = spack.repo.RepoDescriptors.from_config(
+        lock=spack.repo.package_repository_lock(), config=spack.config.CONFIG
+    )
+    for _, desc in descriptors.items():
+        index = desc.path.find("repos/spack_repo")
```

Comment:
the dirs between the git repo root and `spack_repo/<namespace>` can be arbitrary, so this isn't how to find the git repo root from a package path.

Probably easiest to use `spack.repo.PATH.repo_for_pkg(<name>)` to get a repo instance, use its root, and then ask git what the git root is or `os.path.dirname` until `.git` exists as a subdir.

Reply:
I figured it would be an issue but wasn't sure how general to be. Thanks!

Reply:
🤔 BUT that isn't going to help if I want to get spack to show me the blame for a non-package file in a spack repository.

Reply:
Switched to get the repo root for a python package and updated the search to look for `.git` if the argument is a file.

Reply:
You can assume it's in the same directory as `.git`

Reply:
Resolving comment since no longer relevant.